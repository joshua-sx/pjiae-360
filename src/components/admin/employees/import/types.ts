export interface EmployeeImportData {
  uploadMethod: 'upload' | 'paste' | 'manual' | null;
  csvData: {
    rawData: string;
    headers: string[];
    rows: string[][];
    columnMapping: Record<string, string>;
  };
  uploadedFile: { name: string; size: number } | null;
  manualEmployees: EmployeeData[];
}

export interface EmployeeData {
  employeeId: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  jobTitle: string;
  department: string;
  division: string;
}

export type ImportStep = 'upload' | 'mapping' | 'preview' | 'importing';