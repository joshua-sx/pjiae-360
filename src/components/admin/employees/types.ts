export type { Employee } from "@/types/shared";

export interface EmployeeFilters {
  search: string;
  status: string;
  role: string;
  division: string;
  department: string;
}
