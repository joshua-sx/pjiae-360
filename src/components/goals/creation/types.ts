import { Employee } from '../types';

export interface GoalData {
  title: string;
  description: string;
  assignee: string;
  selectedEmployee: Employee | null;
  dueDate: Date | undefined;
  priority: string;
  type: 'individual' | 'team';
}

export interface GoalCreationStep {
  title: string;
  subtitle: string;
  fields: (keyof GoalData)[];
}