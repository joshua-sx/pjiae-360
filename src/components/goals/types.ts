
export interface Goal {
  id: string;
  title: string;
  employeeName: string;
  employeeId: string;
  dueDate: string | null;
  status: "draft" | "active" | "completed" | "cancelled";
  description?: string;
  type: string;
  weight: number;
  year: string;
}

export interface DivisionGoal {
  title: string;
  description: string;
  director: string;
  directorTitle: string;
  progress: number;
  status: "Not Started" | "In Progress" | "Completed" | "At Risk";
  helperText: string;
}

export interface Employee {
  id: string;
  name: string;
  role: string;
  department: string;
  avatar?: string;
}

export interface Team {
  id: string;
  name: string;
  department: string;
  memberCount: number;
}

export interface NewGoalData {
  scope: "individual" | "team";
  assignedTo: string;
  assignedToName: string;
  title: string;
  description: string;
  successCriteria: string;
  dueDate: Date | undefined;
  priority: string;
  tags: string[];
}
