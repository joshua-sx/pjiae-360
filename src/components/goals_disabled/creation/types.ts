import { Employee } from '../types';

export interface GoalData {
  title: string;
  description: string;
  assignee: string;
  selectedEmployee: Employee | null;
  selectedEmployees: Employee[];
  dueDate: Date | undefined;
  priority: string;
}

export interface GoalCreationStep {
  title: string;
  subtitle: string;
  fields: (keyof GoalData)[];
}