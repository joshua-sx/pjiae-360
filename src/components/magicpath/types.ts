export interface MagicPathEmployee {
  id: string;
  name: string;
  role: string;
  department: string;
  avatar?: string;
}

export interface MagicPathGoalData {
  selectedEmployees: MagicPathEmployee[];
  goalName: string;
  description: string;
  startDate: string;
  endDate: string;
  priority: 'Low' | 'Medium' | 'High';
  metrics: string[];
}

export interface MagicPathGoalCreatorProps {
  onComplete?: (goalData: MagicPathGoalData) => void;
}