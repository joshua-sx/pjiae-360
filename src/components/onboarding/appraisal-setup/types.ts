export interface Competency {
  name: string;
  description: string;
  optional?: boolean;
  applicable?: boolean;
}

export interface CycleData {
  frequency: "annual" | "bi-annual";
  cycleName: string;
  startDate: string;
  goalSettingWindows: Array<{
    id: string;
    name: string;
    startDate: Date;
    endDate: Date;
  }>;
  reviewPeriods: Array<{
    id: string;
    name: string;
    startDate: Date;
    endDate: Date;
    goalWindowId: string;
  }>;
  competencyCriteria: {
    enabled: boolean;
    model: string;
    customCriteria: string[];
    scoringSystem: string;
    competencies?: Competency[];
  };
  notifications: {
    enabled: boolean;
    email: boolean;
    emailAddress: string;
    reminders: boolean;
    deadlines: boolean;
  };
}

export const defaultCycleData: CycleData = {
  frequency: "annual",
  cycleName: "2024 Annual Performance Review",
  startDate: new Date().toISOString().split('T')[0],
  goalSettingWindows: [{
    id: "gsw-1",
    name: "Q1 Goal Setting",
    startDate: new Date("2024-01-01"),
    endDate: new Date("2024-01-31")
  }],
  reviewPeriods: [{
    id: "rp-1",
    name: "Mid-Year Review",
    startDate: new Date("2024-06-01"),
    endDate: new Date("2024-06-30"),
    goalWindowId: "gsw-1"
  }, {
    id: "rp-2",
    name: "Year-End Review",
    startDate: new Date("2024-12-01"),
    endDate: new Date("2024-12-31"),
    goalWindowId: "gsw-1"
  }],
  competencyCriteria: {
    enabled: true,
    model: "pjiae",
    customCriteria: [],
    scoringSystem: "5-point-scale",
    competencies: [{
      name: "Job Knowledge",
      description: "Understanding of responsibilities, procedures, and required skills."
    }, {
      name: "Quality of Work",
      description: "Accuracy, thoroughness, and attention to detail in tasks."
    }, {
      name: "Productivity",
      description: "Amount of work completed efficiently and effectively."
    }, {
      name: "Initiative",
      description: "Willingness to take on responsibilities, show independence, and go beyond expectations."
    }, {
      name: "Communication Skills",
      description: "Ability to clearly convey information both verbally and in writing."
    }, {
      name: "Teamwork",
      description: "Willingness to cooperate, support team efforts, and contribute to group objectives."
    }, {
      name: "Adaptability/Flexibility",
      description: "Openness to change and ability to adjust to new challenges."
    }, {
      name: "Dependability",
      description: "Reliability in attendance, meeting deadlines, and following through on commitments."
    }]
  },
  notifications: {
    enabled: true,
    email: true,
    emailAddress: "hr@company.com",
    reminders: true,
    deadlines: true
  }
};