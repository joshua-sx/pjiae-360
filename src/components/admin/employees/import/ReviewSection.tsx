import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Alert, AlertDescription } from '@/components/ui/alert';
import { ArrowLeft, ArrowRight, CheckCircle, Mail } from 'lucide-react';
import { EmployeePreviewTable } from '../EmployeePreviewTable';
import { EmployeeData } from './types';

interface ReviewSectionProps {
  employees: EmployeeData[];
  columnMapping: Record<string, string>;
  onBack: () => void;
  onNext: () => void;
}

export function ReviewSection({ employees, columnMapping, onBack, onNext }: ReviewSectionProps) {
  return (
    <div className="space-y-6">
      <Card>
        <CardHeader>
          <CardTitle className="flex items-center gap-2">
            <CheckCircle className="h-5 w-5" />
            Preview & Import
          </CardTitle>
          <CardDescription>
            Review the employee data before importing. Auth users will be created and invitation emails sent.
          </CardDescription>
        </CardHeader>
        <CardContent>
          <div className="space-y-4">
            <Alert>
              <Mail className="h-4 w-4" />
              <AlertDescription>
                Each employee will receive an invitation email with instructions to access their account.
              </AlertDescription>
            </Alert>

            <EmployeePreviewTable employees={employees} columnMapping={columnMapping} />

            <div className="flex justify-end gap-2">
              <Button variant="outline" onClick={onBack}>
                <ArrowLeft className="mr-2 h-4 w-4" />
                Back
              </Button>
              <Button onClick={onNext} disabled={employees.length === 0}>
                <ArrowRight className="mr-2 h-4 w-4" />
                Assign Roles
              </Button>
            </div>
          </div>
        </CardContent>
      </Card>
    </div>
  );
}

