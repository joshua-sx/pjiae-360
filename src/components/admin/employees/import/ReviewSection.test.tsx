import { render } from '@testing-library/react';
import { describe, it, expect, beforeAll } from 'vitest';
import { ReviewSection } from './ReviewSection';
import { EmployeeData } from './types';

describe('ReviewSection', () => {
  beforeAll(() => {
    // jsdom doesn't implement ResizeObserver which is used by DataTable
    (global as any).ResizeObserver = class {
      observe() {}
      unobserve() {}
      disconnect() {}
    };
  });
  it('renders consistently', () => {
    const employees: EmployeeData[] = [];
    const { container } = render(
      <ReviewSection
        employees={employees}
        columnMapping={{}}
        onBack={() => {}}
        onNext={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

