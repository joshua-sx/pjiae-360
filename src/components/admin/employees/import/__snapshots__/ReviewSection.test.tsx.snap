// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewSection > renders consistently 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-4 sm:p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
        >
          <svg
            class="lucide lucide-circle-check-big h-5 w-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.801 10A10 10 0 1 1 17 3.335"
            />
            <path
              d="m9 11 3 3L22 4"
            />
          </svg>
          Preview & Import
        </h3>
        <p
          class="text-sm text-muted-foreground"
        >
          Review the employee data before importing. Auth users will be created and invitation emails sent.
        </p>
      </div>
      <div
        class="p-4 sm:p-6"
      >
        <div
          class="space-y-4"
        >
          <div
            class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground bg-background text-foreground"
            role="alert"
          >
            <svg
              class="lucide lucide-mail h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="16"
                rx="2"
                width="20"
                x="2"
                y="4"
              />
              <path
                d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
              />
            </svg>
            <div
              class="text-sm [&_p]:leading-relaxed"
            >
              Each employee will receive an invitation email with instructions to access their account.
            </div>
          </div>
          <div
            class="space-y-4"
          >
            <div
              class="text-sm text-muted-foreground"
            >
              0 employees ready to import
            </div>
            <div
              class="w-full max-w-full"
            >
              <div
                class="w-full"
              >
                <div
                  class="relative w-full"
                >
                  <div
                    class="rounded-md border w-full overflow-x-auto"
                  >
                    <div
                      class="relative w-full overflow-auto"
                    >
                      <table
                        class="caption-bottom text-sm w-full"
                      >
                        <thead
                          class="[&_tr]:border-b"
                        >
                          <tr
                            class="transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted border-b"
                          >
                            <th
                              class="align-middle text-muted-foreground [&:has([role=checkbox])]:pr-0 h-12 px-4 py-3 text-left font-medium text-sm whitespace-nowrap"
                            >
                              Name
                            </th>
                            <th
                              class="align-middle text-muted-foreground [&:has([role=checkbox])]:pr-0 h-12 px-4 py-3 text-left font-medium text-sm whitespace-nowrap"
                            >
                              Email
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="[&_tr:last-child]:border-0"
                        >
                          <tr
                            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                          >
                            <td
                              class="align-middle [&:has([role=checkbox])]:pr-0 p-4 px-4 py-8 text-center"
                              colspan="2"
                            >
                              <div
                                class="flex flex-col items-center justify-center py-12 text-center"
                              >
                                <div
                                  class="mx-auto flex h-12 w-12 items-center justify-center rounded-full bg-muted"
                                >
                                  <svg
                                    class="lucide lucide-search h-6 w-6 text-muted-foreground"
                                    fill="none"
                                    height="24"
                                    stroke="currentColor"
                                    stroke-linecap="round"
                                    stroke-linejoin="round"
                                    stroke-width="2"
                                    viewBox="0 0 24 24"
                                    width="24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <circle
                                      cx="11"
                                      cy="11"
                                      r="8"
                                    />
                                    <path
                                      d="m21 21-4.3-4.3"
                                    />
                                  </svg>
                                </div>
                                <h3
                                  class="mt-4 text-sm font-medium text-foreground"
                                >
                                  No results found
                                </h3>
                                <p
                                  class="mt-2 text-sm text-muted-foreground max-w-sm"
                                >
                                  Try adjusting your filters or search terms
                                </p>
                                <div
                                  class="mt-4 flex items-center gap-2 text-xs text-muted-foreground"
                                >
                                  <svg
                                    class="lucide lucide-filter h-3 w-3"
                                    fill="none"
                                    height="24"
                                    stroke="currentColor"
                                    stroke-linecap="round"
                                    stroke-linejoin="round"
                                    stroke-width="2"
                                    viewBox="0 0 24 24"
                                    width="24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <polygon
                                      points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
                                    />
                                  </svg>
                                  <span>
                                    Try clearing filters or searching with different terms
                                  </span>
                                </div>
                              </div>
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex justify-end gap-2"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 touch-manipulation border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 text-sm min-h-[44px] sm:min-h-0 [&_svg]:size-4"
            >
              <svg
                class="lucide lucide-arrow-left mr-2 h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m12 19-7-7 7-7"
                />
                <path
                  d="M19 12H5"
                />
              </svg>
              Back
            </button>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 touch-manipulation bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 text-sm min-h-[44px] sm:min-h-0 [&_svg]:size-4"
              disabled=""
            >
              <svg
                class="lucide lucide-arrow-right mr-2 h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="m12 5 7 7-7 7"
                />
              </svg>
              Assign Roles
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
